app/scripts/lib/stores/persistence-manager.ts

```diff
@@ -140,20 +140,44 @@ export class PersistenceManager {
 
   #localStore: BaseStore;
 
-  #backupDb: IndexedDBStore;
+  #backupDb: IndexedDBStore | null = null;
 
   #backup?: string;
 
   #open: boolean = false;
 
   constructor({ localStore }: { localStore: BaseStore }) {
     this.#localStore = localStore;
-    this.#backupDb = new IndexedDBStore();
   }
 
   async open() {
     if (!this.#open) {
-      await this.#backupDb.open('metamask-backup', 1);
+      try {
+        const db = new IndexedDBStore();
+        await db.open('metamask-backup', 1);
+        this.#backupDb = db;
+      } catch (error) {
+        // `indexedDB` can't be used by addons in FF in some instances of
+        // private browsing mode due to this bug:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1982707. In these
+        // cases we just won't have a backup vault.
+        if (
+          isObject(error) &&
+          error instanceof DOMException &&
+          error.name === 'InvalidStateError' &&
+          error.message ===
+            'A mutation operation was attempted on a database that did not allow mutations.'
+        ) {
+          captureException(error);
+          console.warn(
+            'Could not open backup database; automatic vault recovery will not be available.',
+          );
+          console.error(error);
```

Comment:
Notice that We do not rethrow here, as we consider this error "handled", even though we didn't really do anything to handle it.